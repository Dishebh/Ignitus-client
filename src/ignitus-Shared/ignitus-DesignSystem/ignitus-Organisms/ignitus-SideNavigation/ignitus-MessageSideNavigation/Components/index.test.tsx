import React from 'react';
import { mount, render } from 'enzyme';

import MessageSideNavigation, { Layer, Notification } from './index';

let wrapper;
beforeEach(() => {
  wrapper = mount(<MessageSideNavigation />);
});

describe('<DefaultButtonWithIcon />', () => {
  it('should rendered', () => {
    const tree = render(<MessageSideNavigation />);
    expect(tree).toMatchSnapshot();
  });

  it('should rendered all layers', () => {
    expect(wrapper.find(Layer).length).toEqual(4);
  });

  it('should rendered all notifications', () => {
    expect(wrapper.find(Notification).length).toEqual(4);
  });

  it('should rendered notification with correct value', () => {
    const size = wrapper.find(Notification).length;
    for (let i = 0; i < size; i += 1)
      expect(
        wrapper
          .find(Notification)
          .at(i)
          .text(),
      ).toEqual('4');
  });
});
