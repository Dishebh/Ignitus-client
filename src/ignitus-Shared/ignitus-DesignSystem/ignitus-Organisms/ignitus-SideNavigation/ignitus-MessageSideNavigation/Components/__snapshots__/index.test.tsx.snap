// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DefaultButtonWithIcon /> should rendered 1`] = `
<div
  class="css-1j1397a"
>
  <div
    class="css-10yz6th"
  >
    <p
      class="css-7f2qlu"
      color="#000066"
    >
      Direct Messages
    </p>
     
    <div
      class="css-86xyh8"
    >
      4
    </div>
  </div>
  <div
    class="css-td5ga5"
  >
    <p
      class="css-7f2qlu"
      color="#000066"
    >
      Mentors
    </p>
     
    <div
      class="css-86xyh8"
    >
      4
    </div>
  </div>
  <div
    class="css-td5ga5"
  >
    <p
      class="css-7f2qlu"
      color="#000066"
    >
      Organizations
    </p>
     
    <div
      class="css-86xyh8"
    >
      4
    </div>
  </div>
  <div
    class="css-10tn20j"
  >
    <p
      class="css-7f2qlu"
      color="#000066"
    >
      Groups
    </p>
     
    <div
      class="css-86xyh8"
    >
      4
    </div>
  </div>
</div>
`;
